sqlmesh/core/state_sync/common.py

```diff
@@ -106,12 +115,21 @@ def promote(
         }
 
         table_infos = set(environment.promoted_snapshots)
-        if existing_environment and existing_environment.finalized_ts:
+        if (
+            existing_environment
+            and existing_environment.finalized_ts
+            and not environment_suffix_target_changed
+        ):
             # Only promote new snapshots.
             table_infos -= set(existing_environment.promoted_snapshots)
 
         self._update_environment(environment)
-        return list(table_infos), [existing_table_infos[name] for name in missing_models]
+        removed = (
+            list(existing_table_infos.values())
+            if environment_suffix_target_changed
+            else [existing_table_infos[name] for name in missing_models]
+        )
+        return list(table_infos), (removed, removed_environment_naming_info)
```

Comment:
i know you didn't change this but let's do sorted(table_infos) instead of list() so it's deterministic